ruler: Make alertmanager client and all related config per-tenant configurable (#10816)
grafana/mimir: docs/sources/mimir/configure/configuration-parameters/index.md

[notification_timeout: <duration> | default = 10s]

alertmanager_client:
  # (advanced)

>   [tls_enabled: <boolean> | default = ]

Author: Since we unbound the flags for the old ruler config fields, we also lose the docstrings. It's just moved down below under limits. That seems to be desired/expected by the process though. I chose not to publicly mark these as deprecated just yet.

Reviewer: Personally I do not see a reason not to mark as deprecated. It's another place where people can be warned that it will be deprecated, and if it's going to get moved I would want as much visibility as possible

# CLI flag: -ruler.max-independent-rule-evaluation-concurrency-per-tenant
[ruler_max_independent_rule_evaluation_concurrency_per_tenant: <int> | default = 4]

# Per-tenant Alertmanager client configuration. If not supplied, the tenant's
# notifications are sent to the ruler-wide default.
ruler_alertmanager_client_config:
  # Comma-separated list of URL(s) of the Alertmanager(s) to send notifications
  # to. Each URL is treated as a separate group. Multiple Alertmanagers in HA
  # per group can be supported by using DNS service discovery format,
  # comprehensive of the scheme. Basic auth is supported as part of the URL.

>   # CLI flag: -ruler.alertmanager-url

Author: Here's the result of following the process. It's still a sub-block under limits in yaml, but each option also has its own flag, and it's exactly the set of flags that already existed. The process made this much cleaner IMO, good work @francoposa!

# CLI flag: -ruler.max-independent-rule-evaluation-concurrency-per-tenant
[ruler_max_independent_rule_evaluation_concurrency_per_tenant: <int> | default = 4]

> # Per-tenant alertmanager client configuration. If not supplied, the tenant's

Reviewer 2: ```suggestion
# Per-tenant Alertmanager client configuration. If not supplied, the tenant's
```

> # notifications will be sent to the ruler-wide default.

Reviewer 2: ```suggestion
# notifications are sent to the ruler-wide default.
```
